NumLib/ODESolver/TimeDiscretization.h

```diff
@@ -114,9 +114,20 @@ class InternalMatrixStorage
 class TimeDiscretization
 {
 public:
+    TimeDiscretization() = default;
+
     //! Sets the initial condition.
     virtual void setInitialState(const double t0, GlobalVector const& x0) = 0;
 
+    /*! Get the relative change of solutions between two successive time steps
+     *  by \f$ e_n = \|u^{n+1}-u^{n}\|/\|u^{n+1}\| \f$.
+     *
+     * \param x         The solution at the current timestep.
+     * \param norm_type The type of global vector norm.
+     */
```

Comment:
Different algorithms have different method to calculate the error ( of the change ) of solutions. This special behaviour can be documented in the class of algorithm.